vllm/spec_decode/spec_decode_worker.py

```diff
@@ -302,6 +312,12 @@ def execute_model(
             broadcast_tensor_dict({}, src=0)
             return []
 
+        if self.proposer_uses_kv_cache:
```

Comment:
nit: preference for always running this code so the state is the same with and without `proposer_uses_kv_caches`. this will reduce branches in code and number of states we have to reason about.

Reply:
since we have small batch sizes, I expect latency hit to be <50µs

Reply:
also suggest moving to own function so reader of this method doesn't have to know these details

Reply:
Done. Always running this logic. Also moved it to its own function.